src/Compilers/Core/Portable/Operations/ForEachLoopOperationInfo.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Operations
+{
+    internal struct ForEachLoopOperationInfo
```

Comment:
Is this something we want to consider exposing publicly? At least some of this info might be useful, like the enumerator info.

Reply:
> Is this something we want to consider exposing publicly? At least some of this info might be useful, like the enumerator info.

That is for IOIperation team to decide. I am not planning to expose any of this information publicly under umbrella of CFG work

---
In reply to: [181822345](https://github.com/dotnet/roslyn/pull/26158#discussion_r181822345) [](ancestors = 181822345)

Reply:
Can you open an issue to track that?

---
In reply to: [181826224](https://github.com/dotnet/roslyn/pull/26158#discussion_r181826224) [](ancestors = 181826224,181822345)

Reply:
> Can you open an issue to track that?

[https://github.com/dotnet/roslyn/issues/26176](https://github.com/dotnet/roslyn/issues/26176)

---
In reply to: [181826481](https://github.com/dotnet/roslyn/pull/26158#discussion_r181826481) [](ancestors = 181826481,181826224,181822345)

Reply:
Adding @heejaechang @ivanbasov @CyrusNajmabadi - I believe this is very useful semantic information, which can be very valuable for foreach loop refactorings written by these folks.